lms/djangoapps/course_api/blocks/permissions.py

```diff
@@ -28,7 +28,14 @@ def can_access_self_blocks(requesting_user, course_key):
     """
     Returns whether the requesting_user can access own blocks.
     """
-    return (
+    user_is_enrolled_or_staff = (  # pylint: disable=consider-using-ternary
         (requesting_user.id and CourseEnrollment.is_enrolled(requesting_user, course_key)) or
         has_access(requesting_user, CourseStaffRole.ROLE, course_key)
     )
+    if user_is_enrolled_or_staff:
+        return user_is_enrolled_or_staff
+    else:
+        try:
+            return is_course_public(get_course(course_key, depth=0))
+        except ValueError:
```

Comment:
Do we need to worry about catching this anywhere else we call this?

Reply:
ValueError is raised by get_course if the course is not found. I don't think this should be caught at the `is_course_public` level so this seems to be the best place. 
I don't see an issue in this case because the main scenario for raising the error here is a missing course and in that case `ACCESS_DENIED` seems to make most sense.

Reply:
Sounds good, just wanted to flag it, as it stood out.